src/routes/ExercisesScreens.tsx

```diff
@@ -47,6 +67,17 @@ const ExercisesScreen = ({ route, navigation }: ExercisesScreenProps): JSX.Eleme
 
   return (
     <Root>
+      <ConfirmationModal
+        visible={isModalVisible}
+        setVisible={setIsModalVisible}
+        text="Ups, this exercise is still unlocked."
+        confirmationButtonText="OK,GOT IT!"
+        cancelButtonText="UNLOCK EXERSCISE"
+        confirmationAction={() => null}>
+        <SmallMessage>
+          you have to complete <BoldText>prerequisite</BoldText> to unlock the next exercise.
```

Comment:
you cannot just write `prerequisite` here, it should show the name of the exercise that needs to be finished. You can determine this with your `CurrentLevel` variable.